easybuild/tools/filetools.py

```diff
@@ -1381,14 +1381,21 @@ def apply_regex_substitutions(paths, regex_subs, backup='.orig.eb'):
                     txt_utf8 = txt.decode(encoding='utf-8', errors='replace')
                     write_file(path, txt_utf8)
 
-                for line_id, line in enumerate(fileinput.input(path, inplace=1, backup=backup_ext)):
-                    for regex, subtxt in compiled_regex_subs:
-                        match = regex.search(line)
-                        if match:
-                            origtxt = match.group(0)
-                            _log.info("Replacing line %d in %s: '%s' -> '%s'", (line_id + 1), path, origtxt, subtxt)
-                        line = regex.sub(subtxt, line)
-                    sys.stdout.write(line)
+                backup_path = path + (backup or '.bak')
+                copy_file(path, backup_path)
+                with _open(backup_path, 'r') as in_file:
+                    with _open(path, 'w') as out_file:
+                        for line_id, line in enumerate(in_file):
+                            for regex, subtxt in compiled_regex_subs:
+                                match = regex.search(line)
+                                if match:
+                                    origtxt = match.group(0)
+                                    _log.info("Replacing line %d in %s: '%s' -> '%s'",
+                                              (line_id + 1), path, origtxt, subtxt)
+                                line = regex.sub(subtxt, line)
+                            out_file.write(line)
+                if not backup:
+                    remove_file(backup_path)
```

Comment:
Ah, ok, so we're not changing behavior after all... But we are reading the file twice though, that can be easily avoided now?

Reply:
Done